src/main/java/games/strategy/engine/framework/lookandfeel/LookAndFeel.java

```diff
@@ -65,26 +74,30 @@
     return substanceLooks;
   }
 
+  /**
+   * Sets the user's preferred Look-And-Feel. If not available, the system Look-And-Feel will be used.
+   *
+   * @throws IllegalStateException If this method is not called from the EDT.
+   */
   public static void setupLookAndFeel() {
-    SwingAction.invokeAndWait(() -> {
-      try {
-        UIManager.setLookAndFeel(ClientSetting.LOOK_AND_FEEL_PREF.value());
-        // FYI if you are getting a null pointer exception in Substance, like this:
-        // org.pushingpixels.substance.internal.utils.SubstanceColorUtilities
-        // .getDefaultBackgroundColor(SubstanceColorUtilities.java:758)
-        // Then it is because you included the swingx substance library without including swingx.
-        // You can solve by including both swingx libraries or removing both,
-        // or by setting the look and feel twice in a row.
-      } catch (final Throwable t) {
-        if (!SystemProperties.isMac()) {
-          try {
-            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
-          } catch (final Exception e) {
-            ClientLogger.logQuietly("Failed to set system look and feel", e);
-          }
+    checkState(SwingUtilities.isEventDispatchThread());
+
+    try {
+      UIManager.setLookAndFeel(ClientSetting.LOOK_AND_FEEL_PREF.value());
+      // FYI if you are getting a null pointer exception in Substance, like this:
+      // org.pushingpixels.substance.internal.utils.SubstanceColorUtilities
+      // .getDefaultBackgroundColor(SubstanceColorUtilities.java:758)
+      // Then it is because you included the swingx substance library without including swingx.
+      // You can solve by including both swingx libraries or removing both,
+      // or by setting the look and feel twice in a row.
```

Comment:
This comment can very likely be removed as the "new" substance library we use is internally very different to the original one.

Reply:
I thought `swingx` may now refer to what's called `trident`.  When I remove Trident from the classpath, I get a `NoClassDefFoundError` instead of an NPE, and the `NoClassDefFoundError` makes it obvious that Trident is missing.

So, I agree that the condition this comment is describing probably no longer occurs with the current Substance library.

Reply:
ef88a4636